### PR TITLE
Pull in latest from upstream with accepted hpa metrics change

### PR DESCRIPTION
**Pull request by ashleycutalo**
Grabbed the metrics we're interested in off of the models here https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/types.go

### CODE REVIEW

**Comment by ashleycutalo** [comment]
meow #compute @paulnivin

**Comment by paulnivin** [comment]
let's include MinReplicas and MaxReplicas?

File: internal/store/hpa.go [hunk 1]
```diff
@@ -143,6 +143,51 @@ var (
...
+		{
+			Name: "kube_hpa_status_current_value",
+			Type: metric.Gauge,
+			Help: "Current metric value observed by the autoscaler.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelValues: []string{strings.ToLower(string(c.Resource.Name))},
+							Value: float64(a.Status.CurrentMetrics.Resource.Current.Value),
```
Comment (paulnivin) [inline internal/store/hpa.go]: CurrentMetrics is an array

File: internal/store/hpa.go [hunk 2]
```diff
@@ -132,8 +132,8 @@ var (
 
 					for j, m := range metrics {
 						metric := m
-						metric.LabelKeys = []string{"condition", "status"}
-						metric.LabelValues = append(metric.LabelValues, string(c.Type))
+						metric.LabelKeys = []string{"condition", "status", "reason"}
+						metric.LabelValues = append(metric.LabelValues, string(c.Type), string(c.Reason))
```
Comment (paulnivin) [inline internal/store/hpa.go]: raw data prior to this change:
```$ curl -s 10.45.31.74:8080/metrics | grep hpa_status_condition | grep dispatchavailability-main
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="true",status="AbleToScale"} 1
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="false",status="AbleToScale"} 0
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="unknown",status="AbleToScale"} 0
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="true",status="ScalingActive"} 1
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="false",status="ScalingActive"} 0
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="unknown",status="ScalingActive"} 0
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="true",status="ScalingLimited"} 1
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="false",status="ScalingLimited"} 0
kube_hpa_status_condition{namespace="dispatchavailability-production",hpa="dispatchavailability-main",condition="unknown",status="ScalingLimited"} 0
```

File: internal/store/hpa.go [hunk 3]
```diff
@@ -132,8 +132,8 @@ var (
```
Comment (paulnivin) [inline internal/store/hpa.go]: addConditionMetrics(c.Status, c.Reason) is needed to pre-generate a matrix of all status w/ reason?

File: internal/store/hpa.go (hunk 1 above)
Comment (ashleycutalo) [reply internal/store/hpa.go]: Fixed.

File: internal/store/hpa.go (hunk 3 above)
Comment (ashleycutalo) [reply internal/store/hpa.go]: Reason is an open-ended string unlike Status so I don't think we can generate a matrix off of it.

**Comment by ashleycutalo** [comment]
meow #compute

**Pull request closed without merging**